### DISCUSSION TITLE
How to configure the L2 ChainID to not use the default 1001?

### DISCUSSION THREAD

**Discussion by bap2pecs** [0xPolygonHermez/zkevm-node#2146] [Q&A] [answered]
I am following https://zkevm.polygon.technology/setup-local-node/ to test running a local zkEVM network.

to set my own L2 ChainID, I looked into `test/docker-compose.yml`, `test/config/test.node.config.toml`, and `test/config/test.genesis.config.json` but couldn't find a place for me to override that.

can anyone help?

**Comment by ToniRamirezM** [comment, marked as answer]
Hello,

The L2ChainID is read by the node from the zkEVM Smart Contract. So it is not directly configurable by config files.

For example in our Testnet you can get the chainID by reading it from the contract:

https://goerli.etherscan.io/address/0xa997cfD539E703921fD1e3Cf25b4c241a27a4c7A#readProxyContract

I hope this helps.

**Reply by avenbreaks** [reply]
ask sir, so basically is the chainID left default 1001 for L2? and as a replacement ChainID is read through the L1 contract? correct me if I'm wrong sir.
